### ISSUE TITLE
Arrays

### ISSUE THREAD

**Issue by duckinator** [fasterthanlime/rock#322]
This was incredibly rude at first (sorry all who read it :\) so I rewrote/reworded it:

Current Array implementation: https://github.com/nddrylliog/rock/blob/master/sdk/lang/Array.h

ooc's Array class is pretty much a paper-thin layer over C arrays, and offers little functionality over that besides tracking their size.

Basic idea I had (taken from a comment below):

Array could just act like what you have atm by default, but if you import a particular file (like, structs/Array or similar) it adds more useful features -- so it stays minimal unless you need it to be useful.

TL;DR: imo, Array literals should be more useful than C arrays (even if only marginally so) without casting to a different type.
